### PR TITLE
Upgrade to OpenTracing 1.5.0

### LINKED ISSUE #52: Support newer dynamic loading interface
Could you guys upgrade to the newer dynamic loading interface?

I'm looking to make a new release for nginx-opentracing using OpenTracing-1.5.

You can see https://github.com/lightstep/lightstep-tracer-cpp/pull/105 for an example of how the changes can be made.

### PR DESCRIPTION
**Pull request by willgittoes-dd**
And more substantially; change CI to build nginx-opentracing ourselves.

Otherwise there's a circular dependency where we can't upgrade until they do.

fixes #52

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,3 +36,6 @@
 
 # Bazel
 bazel-*
+
+# IDE settings
```
Comment (cgilmour) [inline .gitignore]: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore so you don't need to do it for every repo your IDE ever touches.

File: src/dynamic_load.cpp [hunk 2]
```diff
@@ -2,20 +2,31 @@
...
+                                         void** tracer_factory) try {
+  if (opentracing_version == nullptr || opentracing_abi_version == nullptr ||
+      error_message == nullptr || error_category == nullptr || tracer_factory == nullptr) {
+    std::cerr << "opentracing_version, opentracing_abi_version, error_message, `error_category, "
+                 " and tracer_factory must be non-null."
+              << std::endl;
+    std::terminate();
+  }
+
+  if (std::strcmp(opentracing_abi_version, OPENTRACING_ABI_VERSION) != 0) {
```
Comment (cgilmour) [inline src/dynamic_load.cpp]: The noise below should probably include ABI versions that weren't matching.

File: src/dynamic_load.cpp [hunk 3]
```diff
@@ -2,20 +2,31 @@
 #include <iostream>
 #include "tracer.h"
 #include "tracer_factory.h"
-#include "version_check.h"
 
-int OpenTracingMakeTracerFactory(const char* opentracing_version, const void** error_category,
-                                 void** tracer_factory) try {
-  if (!datadog::opentracing::equal_or_higher_version(std::string(opentracing_version),
```
Comment (cgilmour) [inline src/dynamic_load.cpp]: Yay, another hack deprecated.

File: test/integration/nginx/Dockerfile [hunk 4]
```diff
@@ -59,7 +73,7 @@ RUN mkdir -p /var/www/
 COPY ./test/integration/nginx/index.html /var/www/index.html
 
 # Get Wiremock
-ADD http://repo1.maven.org/maven2/com/github/tomakehurst/wiremock-standalone/2.18.0/wiremock-standalone-2.18.0.jar wiremock-standalone-2.18.0.jar
+RUN wget http://repo1.maven.org/maven2/com/github/tomakehurst/wiremock-standalone/2.18.0/wiremock-standalone-2.18.0.jar -O wiremock-standalone-2.18.0.jar
```
Comment (cgilmour) [inline test/integration/nginx/Dockerfile]: Any special reason for this?

File: src/dynamic_load.cpp (hunk 2 above)
Comment (willgittoes-dd) [reply src/dynamic_load.cpp]: Still not entirely certain if the old `opentracing_version` is important as well, but from other examples it doesn't seem to be worth checking

File: test/integration/nginx/Dockerfile (hunk 4 above)
Comment (willgittoes-dd) [reply test/integration/nginx/Dockerfile]: Docker will cache `RUN wget` but will execute every single ADD (and check the checksum to see whether to bust the cache). This makes it slow, and makes me feel bad for the webmaster of whoever serves me files.

So for files that I expect to be immutable, RUN wget is nicer.

**Pull request merged**